pkg/plugins/policies/core/rules/resourcerules.go

```diff
@@ -0,0 +1,257 @@
+package rules
+
+import (
+	"fmt"
+	"strings"
+
+	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	meshextenralservice_api "github.com/kumahq/kuma/pkg/core/resources/apis/meshexternalservice/api/v1alpha1"
+	meshservice_api "github.com/kumahq/kuma/pkg/core/resources/apis/meshservice/api/v1alpha1"
+	core_model "github.com/kumahq/kuma/pkg/core/resources/model"
+	meshhttproute_api "github.com/kumahq/kuma/pkg/plugins/policies/meshhttproute/api/v1alpha1"
+)
+
+type ResourceRule struct {
+	Resource            core_model.ResourceMeta
+	ResourceSectionName string
+	Conf                []interface{}
+	Origin              []Origin
+}
+
+type Origin struct {
+	Resource core_model.ResourceMeta
+	// RuleIndex is an index in the 'to[]' array, so we could unambiguously detect what to-item contributed to the final conf.
+	// Especially useful when to-item uses `targetRef.Labels`, because there is no obvious matching between the specific resource
+	// in `ResourceRule.Resource` and to-item.
+	RuleIndex int
+}
+
+type UniqueResourceIdentifier struct {
+	core_model.ResourceIdentifier
+
+	ResourceType core_model.ResourceType
+	SectionName  string
+}
+
+func (ri UniqueResourceIdentifier) String() string {
+	var pairs []string
```

Comment:
would we need to create ResourceIdentifier from this string? When I started implementing InspectAPI I needed to extract resource type, so some part of it. Now resource type can be empty, should we always add something?

Reply:
I actually changed the `ResourceRules` map to have `UniqueResourceIdentifier` as a key. So there is no need now to parse it